### DISCUSSION TITLE
Sidecar Plugin (CMPv2) - Unclean work area - git stash needed

### DISCUSSION THREAD

**Discussion by thedatabaseme** [argoproj/argo-cd#13986] [General]
Hello,

I'm using an Argo plugin to do several things with my application repo data before deployment. One of it is I need to `git-crypt unlock` the repo. Since I switched from configmap plugins to sidecar plugins, I experienced an error during plugin execution which comes from git-crypt. 

```
Error: Working directory not clean.
Please commit your changes or 'git stash' them before running 'git-crypt unlock'.
```

So I had to workaround it by doing a `git stash` in the plugin.

So I thought, I don't really have a clue how git checkout / cloning works in Argo when running a sidecar plugin. I checked the documentation, but I haven't found any particular good explanation on that topic. Long story short, maybe someone can explain / answer the following questions to me:

- How does git checkout work with a sidecar plugin? Does the repo-server container do the checkout or the sidecar?
- If the repo-server does it, how the actual repo data is transfered to the sidecar so that the plugin can do stuff with it?
- Is there a documentation on the actual "how it works" that I've missed?
- Why thereby the git work area becomes "dirty" / not clean? There should be no changes in the repo data involved?!
- Is there a known way to prevent a dirty work area?

Kind regards
Philip

**Comment by cristian-trombetta-fincons** [comment]
Happens also to me, so annoying
